core/storage/btree.rs

```diff
@@ -5508,7 +5641,7 @@ impl CursorTrait for BTreeCursor {
                         loop {
                             if !self.stack.has_parent() {
                                 // All pages of the b-tree have been visited. Return successfully
-                                let c = self.move_to_root()?;
+                                let c = return_if_io!(self.move_to_root());
```

Comment:
How does this work? If we yield on IO here, then `count_state` won't be updated. If we don't yield on IO here, then is `c` ever `Some`?